pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -250,3 +279,61 @@ func newTestHTTPClient(response *http.Response, err error) *http.Client {
 		},
 	}
 }
+
+type benchmarkingRoundTripper struct {
+	dataType benchmarkingDataType
+}
+
+func (brt benchmarkingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	return makeProtoResponseForBenchmarking(http.StatusOK, brt.dataType, network.ProtoAcceptContent), nil
```

Comment:
This includes the allocations needed to make this response to be part of each and every measured operation, skewing the results imo.

Should we instead allocate the response once and only rewind the body's reader in between operations?

Reply:
It does skew the results but what matters is relative numbers. Of course we can measure exactly what we care about is better.